### PR TITLE
Add C FFI bindings for timelock encryption

### PR DESCRIPTION
**Pull request by TrustMeItsCugel**
This PR introduces C/C++ bindings for the timelock encryption library, enabling native integration in C/C++ projects.

### Features
- Complete C API with encrypt/decrypt operations and Drand integration
- Memory-safe interface with proper error handling and cleanup
- Cross-platform support (Windows, Linux, macOS)
- Automatic header generation via cbindgen
- Working examples in both C and C++

### What's Added
- timelock-ffi - New FFI crate with comprehensive test suite (24 tests)
- timelock-ffi - C/C++ examples with CMake and Make build systems
- Cross-platform CI/CD workflow for automated testing
- Complete documentation and build guides

### Testing
- 25 FFI unit tests covering edge cases and memory safety
- 15 core library tests passing
- Working C and C++ examples with Drand beacon decryption
- Cross-platform CI testing on Ubuntu, macOS, and Windows

### Distribution
- Small footprint: ~476KB DLL + 5.6KB header file
- No external dependencies: Self-contained with standard system libraries
- pkg-config template included for future integration

Cross platform CI/CD: https://github.com/TrustMeItsCugel/timelock/actions/

### CODE REVIEW

**Comment by driemworks** [comment]
@TrustMeItsCugel Thanks for the contribution! It's looking really great so far, I appreciate how thorough your updates are. Once the last few copilot comments are addressed, I'll be happy to merge!

**Comment by TrustMeItsCugel** [comment]
Hi @driemworks , I finally got copilot to stop adding comments! Obviously there is a lot of AI in this and I want to do a final review of everything myself. But otherwise I think it is pretty close. Please feel free to add comments, questions, discussion, etc.

File: timelock-ffi/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+    let secret_key_slice = slice::from_raw_parts(secret_key, 32);
+    
+    // Convert secret key to array
+    let mut secret_key_array = [0u8; 32];
+    secret_key_array.copy_from_slice(secret_key_slice);
+
+    // Parse public key hex string
+    let public_key_cstr = match CStr::from_ptr(public_key_hex).to_str() {
+        Ok(s) => s,
+        Err(e) => {
```
[the remark this answers is not in the document]
Comment (TrustMeItsCugel) [reply timelock-ffi/src/lib.rs:317]: This looks like a copilot mistake. It is being zeroized.

**Review by driemworks — APPROVED** [review]
LGTM

File: timelock-ffi/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+// BLS12-381 curve element sizes, as defined by the BLS12-381 specification:
+// - G1 compressed point: 48 bytes  
+// - G2 compressed point: 96 bytes
+// See: RFC 9380, Section 4.3.3
+// (https://datatracker.ietf.org/doc/html/rfc9380#section-4.3.3)
+// These constants are also validated in tests.rs and at runtime to ensure they match
+// the actual compressed_size() values returned by the ark-bls12-381 library.
+// If the library changes its serialization format, the tests will fail,
+// alerting us to update these values.
+pub const BLS_G1_SIZE: usize = 48; // G1 compressed point size (BLS12-381 spec)
```
Comment (driemworks) [inline timelock-ffi/src/lib.rs:59]: It's fine as is, but we can also reference these constants from the specific EngineBLS impl that is being used (https://github.com/ideal-lab5/timelock/blob/277e8d7eaf205e05e2693e0fde8eb2fdd75e45d5/timelock/src/engines/drand.rs#L84). Right now we only officially support BLS12-381, so these values are *fine* and won't be subjected to change if we want to support additional curves later on. At some point, we will likely need to fetch these constants from the [EngineBLS](https://github.com/ideal-lab5/timelock/blob/277e8d7eaf205e05e2693e0fde8eb2fdd75e45d5/timelock/src/engines/engine.rs#L94).

Comment (TrustMeItsCugel) [reply timelock-ffi/src/lib.rs]: Thanks @drieworks

Second latest commit:
- Replace hardcoded BLS sizes with engine-derived constants
- BLS_G1_SIZE now references TinyBLS381::SIGNATURE_SERIALIZED_SIZE
- BLS_G2_SIZE now references TinyBLS381::PUBLICKEY_SERIALIZED_SIZE
- TIMELOCK_SECRET_KEY_SIZE uses engine SECRET_KEY_SIZE constant

Latest commit was just to remove copilot_instructions.md

Are you OK with the CI/CD YML staying in?
